pinot-server/src/main/java/org/apache/pinot/server/api/resources/MmapDebugResource.java

```diff
@@ -45,4 +59,33 @@
   public List<String> getOffHeapSizes() {
     return PinotDataBuffer.getBufferInfo();
   }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/tables/{tableName}/memoryConsumedRealtime")
+  @ApiOperation(value = "Show off heap memory consumed by latest mutable segment", notes = "Returns off heap memory consumed by latest consuming segment of realtime table")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error"), @ApiResponse(code = 404, message = "Table not found")})
+  public String getTableSize(
+      @ApiParam(value = "Table Name with type", required = true) @PathParam("tableName") String tableName)
+      throws WebApplicationException {
+    double memoryConsumed = 0;
+    TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableName);
+    if (tableType != TableType.REALTIME) {
+      throw new WebApplicationException("This api cannot be used with non real-time table: " + tableName,
+          Response.Status.BAD_REQUEST);
+    }
+
+    InstanceDataManager instanceDataManager = serverInstance.getInstanceDataManager();
+    if (instanceDataManager == null) {
+      throw new WebApplicationException("Invalid server initialization", Response.Status.INTERNAL_SERVER_ERROR);
+    }
+    RealtimeTableDataManager realtimeTableDataManager =
+        (RealtimeTableDataManager) instanceDataManager.getTableDataManager(tableName);
+    if (realtimeTableDataManager == null) {
+      throw new WebApplicationException("Table: " + tableName + " is not found", Response.Status.NOT_FOUND);
+    }
+
+    memoryConsumed = realtimeTableDataManager.getStatsHistory().getLatestSegmentMemoryConsumed();
+    return ResourceUtils.convertToJsonString(memoryConsumed);
```

Comment:
This will return a string instead of a json (123 -> "123").
We should create a json to wrap the value.
E.g. `return ResourceUtils.convertToJsonString(Collections.singletonMap("offheapMemoryConsumed", memoryConsumed));`